assets/course-theme/contact-teacher.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { onWindowLoad } from '../shared/helpers/DOM';
+
+function handleSubmit( ev ) {
+	ev.preventDefault();
+	const form = ev.target;
+	const submitButton = ev.target.querySelector(
+		'button.sensei-contact-teacher-form__submit'
+	);
+	submitButton.classList.add( 'sensei-course-theme__button', 'is-busy' );
+	submitButton.disabled = true;
+
+	const fieldNames = [
+		'sensei_message_teacher_nonce',
+		'post_id',
+		'contact_message',
+	];
+	const values = fieldNames.reduce(
+		( acc, name ) => ( {
+			...acc,
+			[ name ]: form.querySelector( `[name="${ name }"]` ).value,
+		} ),
+		{}
+	);
+
+	apiFetch( {
```

Comment:
It didn't work with `window.fetch` on my initial attempts and I resorted using the `@wordpress/api-fetch` instead.

Now that I did some more digging and learning the WP rest api a little more I managed it to work with window.fetch api. c3ce9857424a1e98b3e7499631c4ce56522d42ec

Reply:
Nice! There are various cases where the rest endpoint might not be at `/wp-json` (like multi-sites with a path prefix), so that bit needs an extra step. I found this handy hack to get it from the link tag Wordpress puts out:
`const restUrl = [document.querySelector('link[rel="https://api.w.org/"]').href;`
Probably a good idea to also bail if that's not found, as the REST API can also be disabled entirely.

Reply:
This is great! Didn't know about that trick. Thanks for the tip.

Also, now since the modal opens the clone of the modal content instead of the original element, I updated the `contact-teacher.js` to address the lost event handlers. 2cc53feb8dbd486741049c5f3c43556954f1a1a3